google-cloud-spanner/src/test/java/com/google/cloud/spanner/BenchmarkingDataLoaderUtility.java

```diff
@@ -0,0 +1,173 @@
+ * Hosts a bunch of utility methods/scripts that can be used while performing benchmarks to load
+ * data, report latency metrics, etc.
+ */
+@Category(SlowTest.class)
+@RunWith(JUnit4.class)
+public class BenchmarkingDataLoaderUtility {
+  @ClassRule public static IntegrationTestEnv env = new IntegrationTestEnv();
+  private static final String PROJECT_ID = "";
+  private static final String INSTANCE_ID = "";
+  private static final String DATABASE_ID = "";
+  private static final String SERVER_URL = "https://staging-wrenchworks.sandbox.googleapis.com";
+  private static DatabaseClient client;
+  private static Spanner spanner;
+
+  @BeforeClass
+  public static void beforeClass() {
+    final SpannerOptions.Builder optionsBuilder =
+        SpannerOptions.newBuilder()
+            .setProjectId(PROJECT_ID)
+            .setAutoThrottleAdministrativeRequests();
+    if (!SERVER_URL.isEmpty()) {
+      optionsBuilder.setHost(SERVER_URL);
+    }
+    final SpannerOptions options = optionsBuilder.build();
+    spanner = options.getService();
+    client = spanner.getDatabaseClient(DatabaseId.of(PROJECT_ID, INSTANCE_ID, DATABASE_ID));
+  }
+
+  @AfterClass
+  public static void afterClass() {
+    spanner.close();
+  }
+
+  /**
+   * A utility which bulk inserts 10^5 records into the database in batches. The method assumes that
+   * the instance/database/table is already created. It does not perform any admin operations.
+   *
+   * <p>Table schema used here: CREATE TABLE FOO ( id INT64 NOT NULL, BAZ INT64, BAR INT64, )
+   * PRIMARY KEY(id);
+   */
+  @Test
+  public void bulkInsertTestData() {
+    String insertQuery = "INSERT INTO FOO(id, BAZ, BAR) VALUES(@key, @val1, @val2)";
+    int key = 0;
+    for (int batch = 0; batch < 1000; batch++) {
+      List<Statement> stmts = new LinkedList<>();
+      for (int i = 0; i < 100; i++) {
+        stmts.add(
+            Statement.newBuilder(insertQuery)
+                .bind("key")
+                .to(key)
+                .bind("val1")
+                .to(1)
+                .bind("val2")
+                .to(2)
+                .build());
+        key++;
+      }
+      final TransactionCallable<long[]> callable = transaction -> transaction.batchUpdate(stmts);
+      TransactionRunner runner = client.readWriteTransaction();
+      long[] actualRowCounts = runner.run(callable);
+      assertThat(actualRowCounts.length).isEqualTo(100);
+      System.out.println("Completed batch => " + batch);
+    }
+  }
+
+  /**
+   * Collects all results from a collection of future objects.
+   *
+   * @param service
+   * @param results
+   * @param numOperations
+   * @return
+   * @throws Exception
+   */
+  public static List<Duration> collectResults(
+      final ListeningScheduledExecutorService service,
+      final List<ListenableFuture<List<Duration>>> results,
+      final int numOperations)
+      throws Exception {
+    service.shutdown();
+    if (!service.awaitTermination(60L, TimeUnit.MINUTES)) {
```

Comment:
Considering that it is a generic and public method: Would it make sense to add the timeout as a method argument, so the caller can determine what the max wait should be?

Reply:
Sure, done